script.js

```diff
@@ -1,4 +1,159 @@
+There are 2 players and players take turns.
+When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+How many game modes:
+1. Player 1 click submit to roll dice.
+2. Player 1 click 1 or 2 to choose order of dice. 
+3. Player 2 click submit to roll dice. 
+4. Player 2 click 1 or 2 to choose order of dice. 
+5. Determine who gets the higher combined number.  
+*/
+
+//global variables for current game mode
+var playerRoll1 = 0;
+var playerRoll2 = 0;
+var gameMode = "start game by rolling 2 dice";
+var currentPlayer = 1;
+var allDiceOrder = [];
+var accumulativeRound = 0;
+
+var playerRollDice = function () {};
+
+// random dice generator for user to roll dice
+var randomDiceGenerator = function () {
+  playerRoll1 = Math.floor(Math.random() * 6 + 1);
+  playerRoll2 = Math.floor(Math.random() * 6 + 1);
+  console.log(`dice1 output:`, playerRoll1, "& dice2 output:", playerRoll2);
+
+  if (playerRoll1 == playerRoll2) {
+    return `Welcome Player ${currentPlayer}.<br> <br> You have rolled ${playerRoll1} for both dice one and dice two. <br><br>Press 1.`;
+  } else
+    return `Welcome Player ${currentPlayer}. <br> <br> You have rolled ${playerRoll1} for dice one and ${playerRoll2} for dice two.<br> <br>Choose the order of the dice by entering "1" or "2".`;
+};
+
+// function for player to roll 2 dice to get 2 number
+var chooseDiceOrder = function (userInput) {
+  orderOfDice = userInput;
+  //input validation > error message for player if they did not enter 1 or 2
+  if (userInput != 1 && userInput != 2) {
+    return "Please enter either 1 or 2 to choose which dice to put infront.";
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+
+  if (userInput == "1") {
+    var diceOrder = Number(String(playerRoll1) + String(playerRoll2));
+    console.log(`user chose ${playerRoll1}, then ${playerRoll2}.`);
+    //game mode switches to the next player
+    gameMode = "next player roll 2 dice";
+    console.log(gameMode);
+    myOutputValue = `Player ${currentPlayer}, you have chosen dice 1. <br><br>Your final value is ${playerRoll1}${playerRoll2}.<br><br> Please click submit.`;
+  }
+
+  //if user choose 1 then first dice will be infront and second dice will be behind.
+  if (userInput == "2") {
+    diceOrder = Number(String(playerRoll2) + String(playerRoll1));
+    console.log(`user chose ${playerRoll2}, then ${playerRoll1}.`);
+    gameMode = "next player roll 2 dice";
+    //game mode switches to the next player
+    console.log(gameMode);
+    myOutputValue = `You have chosen 2. <br><br>Your final value is ${playerRoll2}${playerRoll1}. <br> <br> Please click submit.`;
+  }
```

Comment:
this is fairly repetitive, notice the only difference between the two `if`s is really just line 48 and 58 so we prob can do that an `if-else`, and everything else is outside of it

```
  var diceOrder;
  if (userInput == "1") {
     diceOrder = Number(String(playerRoll1) + String(playerRoll2));
  } else if (userInput == "2") {
    diceOrder = Number(String(playerRoll2) + String(playerRoll1));
 }

    gameMode = "next player roll 2 dice";
    myOutputValue = `Player ${currentPlayer}, you have chosen dice ${userInput}. <br><br>Your final value is ${diceOrder}.<br><br> Please click submit.`;
```

Reply:
the reason why i emphasise the point of repetition so much is because in programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code.